artman/tasks/package_metadata_tasks.py

```diff
@@ -94,12 +95,42 @@ def _write_yaml(self, config_dict, dest):
         with open(dest, 'w') as f:
             yaml.safe_dump(config_dict, f, default_flow_style=False)
 
-class ProtoPackageMetadataGenTask(task_base.TaskBase):
-    default_provides = 'proto_pkg_dir'
+class JavaGrpcPackageMetadataConfigGenTask(PackageMetadataConfigGenTask):
+    def _create_config(self, api_name, api_version, api_full_name, output_dir,
+                package_dependencies_yaml, package_defaults_yaml, proto_deps,
+                language, local_paths, src_proto_path, package_type,
+                gapic_api_yaml, release_level=None, generated_package_version=None):
+        config = super(JavaGrpcPackageMetadataConfigGenTask, self)._create_config(
+            api_name, api_version, api_full_name, output_dir,
+            package_dependencies_yaml, package_defaults_yaml, proto_deps,
+            language, local_paths, src_proto_path, package_type,
+            gapic_api_yaml, release_level=release_level,
+            generated_package_version=generated_package_version)
+        config['generation_layer'] = 'grpc'
 
-    def execute(self, api_name, api_version, organization_name, local_paths,
+        return config
+
+class JavaProtoPackageMetadataConfigGenTask(PackageMetadataConfigGenTask):
+    def _create_config(self, api_name, api_version, api_full_name, output_dir,
+                package_dependencies_yaml, package_defaults_yaml, proto_deps,
+                language, local_paths, src_proto_path, package_type,
+                gapic_api_yaml, release_level=None, generated_package_version=None):
+        config = super(JavaProtoPackageMetadataConfigGenTask, self)._create_config(
+            api_name, api_version, api_full_name, output_dir,
+            package_dependencies_yaml, package_defaults_yaml, proto_deps,
+            language, local_paths, src_proto_path, package_type,
+            gapic_api_yaml, release_level=release_level,
+            generated_package_version=generated_package_version)
+        config['generation_layer'] = 'proto'
+
+        return config
+
+# Metadata gen
+
+class ProtoPackageMetadataGenTaskBase(task_base.TaskBase):
+    def _execute(self, api_name, api_version, organization_name, local_paths,
                 descriptor_set, src_proto_path, service_yaml,
-                grpc_code_dir, output_dir, package_metadata_yaml,
+                input_dir, output_dir, package_metadata_yaml,
```

Comment:
This `ProtoPackageMetadataGenTaskBase` becomes a base class of `GrpcPackageMetadataGenTask` and `ProtoPackageMetadataGenTask`. The input_dir is either `proto_code_dir` or `grpc_code_dir` depending on the caller